### PR TITLE
feat: LM2-2297 referrals journey

### PR DESCRIPTION
**Pull request by andivido**
This PR is a pretty comprehensive rewrite of the CreditRequest class, which handles Webhooks. Primarily it looks to break the functionality of the class into various methods so that a) it can be customised a little more easily, b) it can be tested a little more easily, c) it's not nearly so impossible to follow/understand.

There are a couple of changes to the functionality:

- Introduces a CREATION_STATUSES const array which triggers the construction of the Order when we receive the appropriate status (in this instance; READY or REFERRED, as outlined [here](https://divido.atlassian.net/browse/LM2-2297?atlOrigin=eyJpIjoiZTE2N2Q4Njk5ODcyNGZhNTk4Mzc4YmJhZTcyM2IwYjQiLCJwIjoiaiJ9))
- Triggers a request to update the merchant reference of an application with the Order ID when a webhook is received that has a status included in the new MERCHANT_REFERENCE_STATUSES const array (currently READY or REFERRED, as outlined [here](https://divido.atlassian.net/jira/software/c/projects/LM2/boards/57?selectedIssue=LM2-2311)) 
- Referred orders are now created (as per point 1) and immediately put into a HOLDED status on the admin panel
- Holded orders are automatically removed from a HOLDED status when a webhook status included in the RELEASE_STATUSES const array is received (currently just the ACCEPTED status)

The rewrite also has the following fixes:
- Requests now return actual json rather than a string of json
- Adds tests for the model
- Removes unnecessary dependencies from the class
- Replaces deprecated save functions for Orders and Quotes

### PR CHECKLIST

- [ ] Ensure any new strings have been translated for import
- [ ] Import new translations via `integrations-magento2`'s `make script-install-languages` command
- [ ] The plugin version (currently as a const in the `Data.php`) has been updated
- [ ] The plugin version in `composer.json` has been updated
- [ ] Tests have been added for any new functionality via `integrations-magento2`'s `make test` command
- [ ] Ensure logging is kept to a minimum, does not include PII, and is set to the pertinent level

### CODE REVIEW

File: Api/CreditRequestInterface.php [hunk 1]
```diff
@@ -18,15 +18,15 @@ public function create();
      * Update an order with results from credit request
      *
      * @api
-     * @return \Magento\Framework\Controller\Result\Json
+     * @return void
      */
     public function update();
 
     /**
      * Update an order with results from credit request
```
Comment (craigastill) [inline Api/CreditRequestInterface.php:26]: CopyPasta comment from above function.

File: Api/CreditRequestInterface.php [hunk 2]
```diff
@@ -18,15 +18,15 @@ public function create();
...
      *
      * @api
-     * @return \Magento\Framework\Controller\Result\Json
+     * @return void
      */
     public function version();
```
Comment (craigastill) [inline Api/CreditRequestInterface.php:31]: Probably `void` return will make sense when the docstring is updated (or me diving through usage), but gut reaction is that a `version()` function would return `int`/`string`.

File: Model/CreditRequest.php [hunk 3]
```diff
@@ -115,83 +178,197 @@ public function create()
...
-     * @return \Magento\Framework\Controller\ResultJson
      */
-    public function update()
-    {
-        $this->logger->info('Application Update - CreditRequest');
-
-        $debug = $this->config->getValue(
-            'payment/divido_financing/debug',
-            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+    public function update() {
```
Comment (craigastill) [inline Model/CreditRequest.php:184]: Are you happy that this function is creating state and firing off webhooks, but has multiple points where an exception will not gracefully role-back/cancel-forwards these changes?

File: Model/CreditRequest.php [hunk 4]
```diff
@@ -115,83 +178,197 @@ public function create()
...
+                    "Could not unhold the order",
+                    ["reason" => $e->getMessage()]
+                );
             }
-            return $this->webhookResponse(false, 'No lookup');
         }
+        
+        $this->orderRepository->save($order);
+
+        return $this->sendWebhookResponse(200, "Order Updated");
```
Comment (craigastill) [inline Model/CreditRequest.php:285]: Worth updating functions return type?

**Review by craigastill — APPROVED** [review]
LGTM.

File: Model/CreditRequest.php (hunk 3 above)
Comment (andivido) [reply Model/CreditRequest.php:184]: This is a problem, but the order save function occurs right at the end of the process, so it shouldn't be too much of an issue. The lookup gets updated midway through, but the updates are only used for internal debugging

File: Api/CreditRequestInterface.php (hunk 1 above)
Comment (andivido) [reply Api/CreditRequestInterface.php]: Good spot

File: Api/CreditRequestInterface.php (hunk 2 above)
Comment (craigastill) [reply Api/CreditRequestInterface.php:32]: Ah, so a setter function, rather than an update&return-response. Cheers!

**Review by craigastill — APPROVED** [review]

**Pull request merged**